### PR TITLE
ArangoDB externs

### PR DESCRIPTION
**Pull request by ciscoheat**
Here are externs for the awesome ArangoDB database: https://www.arangodb.com

API: https://github.com/arangodb/arangojs

Missing most result typedefs, and the `Route` object which currently is Dynamic. Will improve it as I go along.

### CODE REVIEW

File: js/npm/arangojs/ArangoError.hx [hunk 1]
```diff
@@ -0,0 +1,4 @@
+package js.npm.arangojs;
+
+@:native("ArangoError")
```
Comment (clemos) [inline js/npm/arangojs/ArangoError.hx]: It shouldn't be needed ?

File: js/npm/arangojs/Collection.hx [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+}
+
+@:enum abstract CollectionListType(String) to String {
+	var id = "id";
+	var key = "key";
+	var path = "path";
+}
+
+extern class Collection
+//implements npm.Package.Require<"arangodb", "^4.3.0">
```
Comment (clemos) [inline js/npm/arangojs/Collection.hx]: Commented ?

File: js/npm/arangojs/Cursor.hx [hunk 3]
```diff
@@ -0,0 +1,40 @@
+package js.npm.arangojs;
+
+extern class Cursor<T>
+//implements npm.Package.Require<"arangodb", "^4.3.0">
```
Comment (clemos) [inline js/npm/arangojs/Cursor.hx]: ?

Comment (ciscoheat) [inline js/npm/arangojs/Cursor.hx]: Just a leftover, will remove it from all files and update.

File: js/npm/arangojs/ArangoError.hx (hunk 1 above)
Comment (ciscoheat) [inline js/npm/arangojs/ArangoError.hx]: Probably not, nobody should instantiate it anyway. Will remove it.

**Comment by ciscoheat** [comment]
Cleaned up now!

**Comment by clemos** [comment]
Awesome, thanks :)

**Comment by ciscoheat** [comment]
Glad I can help. :) Got some more npm's coming soon.

**Pull request merged**